### DISCUSSION TITLE
How rewrite target working under the hood?

### DISCUSSION THREAD

**Discussion by BinaryWorld2099** [nginx/kubernetes-ingress#3119] [Q&A] [answered]
I was trying the rewrite target in k8s nginx 
here is my example:
nginx.ingress.kubernetes.io/rewrite-target: /$2
path is set to ggg

apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    nginx.ingress.kubernetes.io/rewrite-target: /$2
  name: single-routeggg
  namespace: test
spec:
  ingressClassName: nginx
  rules:
  - host: test4.test5.test6.com
    http:
      paths:
      - path: /ggg(/|$)(.*)
        pathType: Exact
        backend:
          service:
            name: ggg-svc
            port: 
              number: 80  

the service is ExternalName with google.com

apiVersion: v1
kind: Service
metadata:
  name: ggg-svc
  namespace: test
  labels:
    app: ggg
spec:
  type: ExternalName
  externalName: google.com 

the rewrite target is working 
rewrite path form ggg/1/2/3 to /1/2/3

![image](https://user-images.githubusercontent.com/106383882/193442210-623de742-aeaa-47eb-8abf-d429b62be5f1.png)

but checked in Wireshark, the traffic still remain /ggg/1/2/3

![image](https://user-images.githubusercontent.com/106383882/193442323-7b192672-fbed-4974-be7d-de52af79e2e9.png)

What to find out where is the magic trick that change to request to /1/2/3 but not /ggg/1/2/3

**Comment by jasonwilliams14** [comment, marked as answer]
@BinaryWorld2099 Hello there.
In looking at the information you have provided, it looks like you are using annotations for a different NGINX Ingress controller project.
This is the annotation you are using:
```
nginx.ingress.kubernetes.io/rewrite-target: 
```

That is for a different project.
The annotation you will want to use for this project is:

```
nginx.org/rewrites
```

Here is the link to our Ingress controller documentation and the annotations available.

https://docs.nginx.com/nginx-ingress-controller/configuration/ingress-resources/advanced-configuration-with-annotations/#request-uriheader-manipulation

Let us know how it goes and if you have any other questions.

**Comment by BinaryWorld2099** [comment]
Thank you for your time, I am testing in minikube, will try this annotation in future

**Reply by jasonwilliams14** [reply]
@BinaryWorld2099 you are welcome
Let us know how the test goes and if you need any help. We are happy to help out.
